FirebaseInAppMessaging/Swift/Source/CustomInAppMessageDisplayViewModifier.swift

```diff
@@ -0,0 +1,187 @@
+  /// Overrides the default display of a banner in-app message as defined on the Firebase console.
+  func bannerInAppMessage<Content: View>(closure: @escaping (InAppMessagingBannerDisplay,
+                                                             InAppMessagingDisplayDelegate)
+      -> Content)
+    -> some View {
+    modifier(BannerInAppMessageDisplayViewModifier(closure: closure))
+  }
+}
+
+// MARK: Modal messages.
+
+@available(iOS 13, tvOS 13, *)
+struct ModalInAppMessageDisplayViewModifier<DisplayMessage: View>: ViewModifier {
+  var closure: (InAppMessagingModalDisplay, InAppMessagingDisplayDelegate) -> DisplayMessage
+  @ObservedObject var delegateBridge = DelegateBridge.shared
+
+  func body(content: Content) -> some View {
+    return content.overlay(overlayView())
+  }
+
+  @ViewBuilder
+  func overlayView() -> some View {
+    if let modalMessage = delegateBridge.inAppMessageData?.0 as? InAppMessagingModalDisplay,
+      let delegate = delegateBridge.inAppMessageData?.1 {
+      closure(modalMessage, delegate).onAppear { delegate.impressionDetected?(for: modalMessage) }
+    } else {
+      EmptyView()
+    }
+  }
+}
+
+@available(iOS 13, tvOS 13, *)
+public extension View {
+  /// Overrides the default display of a modal in-app message as defined on the Firebase console.
+  func modalInAppMessage<Content: View>(closure: @escaping (InAppMessagingModalDisplay,
+                                                            InAppMessagingDisplayDelegate)
+      -> Content)
+    -> some View {
+    modifier(ModalInAppMessageDisplayViewModifier(closure: closure))
+  }
+}
+
+// MARK: Card messages.
+
+@available(iOS 13, tvOS 13, *)
+struct CardInAppMessageDisplayViewModifier<DisplayMessage: View>: ViewModifier {
+  var closure: (InAppMessagingCardDisplay, InAppMessagingDisplayDelegate) -> DisplayMessage
+  @ObservedObject var delegateBridge = DelegateBridge.shared
+
+  func body(content: Content) -> some View {
+    return content.overlay(overlayView())
+  }
+
+  @ViewBuilder
+  func overlayView() -> some View {
+    if let cardMessage = delegateBridge.inAppMessageData?.0 as? InAppMessagingCardDisplay,
+      let delegate = delegateBridge.inAppMessageData?.1 {
+      closure(cardMessage, delegate).onAppear { delegate.impressionDetected?(for: cardMessage) }
+    } else {
+      EmptyView()
+    }
+  }
+}
+
+@available(iOS 13, tvOS 13, *)
+public extension View {
+  /// Overrides the default display of a card in-app message as defined on the Firebase console.
+  func cardInAppMessage<Content: View>(closure: @escaping (InAppMessagingCardDisplay,
+                                                           InAppMessagingDisplayDelegate)
+      -> Content)
+    -> some View {
+    modifier(CardInAppMessageDisplayViewModifier(closure: closure))
+  }
+}
+
+// MARK: Bridge to Firebase In-App Messaging SDK.
+
+@available(iOS 13, tvOS 13, *)
+class DelegateBridge: NSObject, InAppMessagingDisplay, InAppMessagingDisplayDelegate,
```

Comment:
Nit: can you please add documentation or this since it's the most critical class of the bunch here?

Reply:
Fixed in 99b4a7f.